src/test_l3_plat_power.c

```diff
@@ -0,0 +1,469 @@
+
+     //Step 1: Get the Wakeup Source Type
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Supported Wake Up Source Types");
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("\t#  %-20s","Wakeup Source Type");
+    for(int32_t i = PWRMGR_WAKEUPSRC_VOICE; i <= PWRMGR_WAKEUPSRC_LAN; i++)
+    {
+        UT_LOG_MENU_INFO("\t%d.  %-20s", i, UT_Control_GetMapString(plat_source_types_mapTable, i));
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Select Wake Up Source Type: ");
+    scanf("%d", &sourceType);
+    readAndDiscardRestOfLine(stdin);
+    if(sourceType < PWRMGR_WAKEUPSRC_VOICE || sourceType > PWRMGR_WAKEUPSRC_LAN)
+    {
+        UT_LOG_ERROR("Invalid ARC Type");
+        goto exit;
+    }
+    PWRMGR_WakeupSrcType_t srcType = (PWRMGR_WakeupSrcType_t)sourceType;
+    // Step 2: Call PLAT_API_GetWakeupSrc()
+    UT_LOG_INFO("Calling PLAT_API_GetWakeupSrc()");
+    status = PLAT_API_GetWakeupSrc(srcType,&enable);
+    if ((status != PWRMGR_SUCCESS ))
+    {
+        UT_LOG_INFO("Result PLAT_API_GetWakeupSrc: pmStatus_t:[%s]",
+                 UT_Control_GetMapString(pmStatus_mapTable, status));
+    }
+    else{
+         UT_LOG_MENU_INFO("\t%d.  %-20s is %s", sourceType, UT_Control_GetMapString(plat_source_types_mapTable, sourceType), enable ? "enabled" : "disabled");
+    }
+    UT_ASSERT_EQUAL_FATAL(status, PWRMGR_SUCCESS);
+
+    exit:
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+
+/**
+* @brief Termination of the HAL Power Manager Module
+*
+* This test provides a scope to close the HAL Power Manager module.
+
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 006@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_l3_power_manager_hal_Term(void)
+{
+    gTestID = 6;
+    pmStatus_t status = PWRMGR_SUCCESS ;
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    // Step 1: Call PLAT_TERM()
+    UT_LOG_INFO("Calling PLAT_TERM()");
+    status = PLAT_TERM();
+    UT_LOG_INFO("Result PLAT_TERM: pmStatus_t:[%s]",
+                 UT_Control_GetMapString(pmStatus_mapTable, status));
+
+    UT_ASSERT_EQUAL_FATAL(status, PWRMGR_SUCCESS);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+int test_l3_plat_power_register(void)
+{
+    // Create the test suite
+    pSuite = UT_add_suite("[L3 power manager Functions] ", NULL, NULL);
+    UT_ASSERT_EQUAL_FATAL(pSuite, UT_KVP_STATUS_SUCCESS);
+    // List of test function names and strings
+    
+    UT_add_test( pSuite, "L3_Init_pwrmgr", test_l3_power_manager_hal_Init);
```

Comment:
update this UT_add_test( pSuite, "L3_Init_pwrmgr", test_l3_power_manager_hal_Init);
to 
UT_add_test( pSuite, "Initialize power manger", test_l3_power_manager_hal_Init);
don't use L3 because these menu need used for L4 also

Reply:
Removed L3, and updated the other tests.